### PR TITLE
Refactor route table ID parsing

### PR DESCRIPTION
**Pull request by tyll**
Simplify the parsing of route table IDs to avoid unnecessary exception handling that makes the code harder to understand and triggers CodeQL warnings.

Related to #565

### CODE REVIEW

**Comment by tyll** [comment]
Proper code coverage requires https://github.com/linux-system-roles/.github/pull/11 and https://github.com/linux-system-roles/tox-lsr/pull/109 to be merged and this repo be updated with the changes.

**Comment by richm** [comment]
> Proper code coverage requires [linux-system-roles/.github#11](https://github.com/linux-system-roles/.github/pull/11) 

merged - also submitted https://github.com/linux-system-roles/network/pull/569 for the network role

and [linux-system-roles/tox-lsr#109](https://github.com/linux-system-roles/tox-lsr/pull/109) to be merged and this repo be updated with the changes.

merged - released tox-lsr 2.13.2 - submitted https://github.com/linux-system-roles/network/pull/570

I suggest merging #570 first, then I will rebase #569, then once that is merged, you can rebase this PR

**Comment by tyll** [comment]
Thanks, I merged your PRs and rebased this one.

File: module_utils/network_lsr/argument_validator.py [hunk 1]
```diff
@@ -291,7 +291,6 @@ def _validate_impl(self, value, name):
                     "route table value is {0} but cannot be greater than 4294967295".format(
                         value
                     ),
-                )
```
Comment (liangwen12year) [inline module_utils/network_lsr/argument_validator.py:294]: Spurious removal of this line, but it was added back in another commit.

File: module_utils/network_lsr/argument_validator.py [hunk 2]
```diff
@@ -2639,7 +2639,13 @@ def _parse_route_tables_mapping(cls, file_content, mapping):
...
         # https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/tree/lib/rt_names.c?id=11e41a635cfab54e8e02fbff2a03715467e77ae9#n447
         regex = re.compile(
-            b"^\\s*(0x[0-9a-fA-F]+|[0-9]+)\\s+([a-zA-Z0-9_.-]+)(\\s*|\\s+#.*)$"
+            b"^\\s*"  # optional leading whitespace
+            b"(0x[0-9a-fA-F]+|[0-9]+)"  # table ID in hex or decimal (non negative)
+            b"\\s+"  # whitespace
+            b"("  # make a pattern group for the table name
+            + cls._ROUTE_TABLE_NAME_PATTERN.encode("ascii")
+            + b")"  # close pattern group
+            b"(\\s*|\\s+#.*)$"  # trailing whitespace or comment
```
Comment (liangwen12year) [inline module_utils/network_lsr/argument_validator.py:2649]: Thanks for writting the detailed comment, this is fabulous and much easier to recognize the regex pattern.

File: module_utils/network_lsr/argument_validator.py (hunk 1 above)
Comment (liangwen12year) [reply module_utils/network_lsr/argument_validator.py:294]: Other than this nitpicking, the PR looks good to me.

Comment (tyll) [reply module_utils/network_lsr/argument_validator.py:294]: Thanks, fixed it.

File: module_utils/network_lsr/argument_validator.py (hunk 2 above)
Comment (tyll) [reply module_utils/network_lsr/argument_validator.py:2649]: Thank you.

**Comment by tyll** [comment]
Markdownlint is broken: https://github.com/actionshub/markdownlint/issues/35

**Review by thom311 — APPROVED** [review]

**Review by ffmancera — APPROVED** [review]

**Review by liangwen12year — APPROVED** [review]
LGTM.

**Pull request merged**